src/scorebook/eval_datasets/eval_dataset.py

```
@@ -295,54 +452,128 @@ def from_huggingface(
-        - label: The field used as the evaluation label
+        The YAML file should contain configuration for loading a dataset from Hugging Face.
+
+        Required fields:
+        - path: Hugging Face dataset path
+        - name: Name for the evaluation dataset
         - metrics: List of metrics to evaluate
-        - split: Optional split name to load
-        - template: Optional prompt template
+
+        Input/Label specification (mutually exclusive options):
+        - Direct field names:
+            input: "question"
+            label: "answer"
+        - Templates for constructing from multiple fields:
+            templates:
+              input: "{{ question }}\nOptions: {{ options }}"
+              label: "{{ answer }}"
+
+        Optional fields:
+        - split: Dataset split to load (e.g., "test")
+        - config: Dataset configuration name
 
         Returns:
             An EvalDataset instance configured according to the YAML file.
 
         Raises:
             ValueError: If the YAML file is invalid or missing required fields.
         """
-        path = validate_path(file_path, expected_suffix=".yaml")
+        validated_path = validate_path(path, expected_suffix=(".yaml", ".yml"))
 
         try:
-            with path.open("r", encoding="utf-8") as f:
-                config = yaml.safe_load(f)
+            with validated_path.open("r", encoding="utf-8") as f:
+                yaml_config = yaml.safe_load(f)
         except yaml.YAMLError as e:
-            raise ValueError(f"Invalid YAML in {file_path}: {e}") from e
+            raise ValueError(f"Invalid YAML in {path}: {e}") from e
 
         # Validate required fields
-        required_fields = ["name", "label", "metrics"]
-        missing_fields = [field for field in required_fields if field not in config]
+        required_fields = ["path", "name", "metrics"]
+        missing_fields = [field for field in required_fields if field not in yaml_config]
```

Suggestion: we should also validate metrics (fail fast)

Something like

```python
  # Validate metrics exist before calling from_huggingface
  for metric in yaml_config["metrics"]:
      try:
          MetricRegistry.get(metric)
      except Exception as e:
          raise ValueError(f"Invalid metric '{metric}' in YAML config: {e}")
```

Would work?